osu.Game/Screens/Ranking/Expanded/Statistics/PerformanceStatistic.cs

```diff
@@ -22,30 +29,91 @@ public class PerformanceStatistic : StatisticDisplay
+        private BeatmapManager beatmapManager { get; set; }
+
         public PerformanceStatistic(ScoreInfo score)
             : base("PP")
         {
             this.score = score;
         }
 
         [BackgroundDependencyLoader]
-        private void load(ScorePerformanceCache performanceCache)
+        private void load()
         {
-            if (score.PP.HasValue)
+            Task.WhenAll(
+                // actual performance
+                performanceCache.CalculatePerformanceAsync(score, cancellationTokenSource.Token),
+                // performance for a perfect play
+                getPerfectPerformance(score)
+            ).ContinueWith(attr =>
             {
-                setPerformanceValue(score.PP.Value);
-            }
-            else
+                PerformanceAttributes[] result = attr.GetResultSafely();
+                setPerformanceValue(new PerformanceBreakdown { Performance = result[0], PerfectPerformance = result[1] });
+            });
+        }
+
+        private async Task<PerformanceAttributes> getPerfectPerformance(ScoreInfo originalScore)
+        {
+            ScoreInfo perfectScore = await getPerfectScore(originalScore).ConfigureAwait(false);
+            return await performanceCache.CalculatePerformanceAsync(perfectScore, cancellationTokenSource.Token).ConfigureAwait(false);
+        }
+
+        private Task<ScoreInfo> getPerfectScore(ScoreInfo originalScore)
+        {
+            return Task.Factory.StartNew(() =>
             {
-                performanceCache.CalculatePerformanceAsync(score, cancellationTokenSource.Token)
-                                .ContinueWith(t => Schedule(() => setPerformanceValue(t.GetResultSafely())), cancellationTokenSource.Token);
-            }
+                var beatmap = beatmapManager.GetWorkingBeatmap(originalScore.BeatmapInfo).GetPlayableBeatmap(originalScore.Ruleset, originalScore.Mods);
+                ScoreInfo perfectPlay = originalScore.DeepClone();
+                perfectPlay.Accuracy = 1;
+                perfectPlay.Passed = true;
+
+                // create statistics assuming all hit objects have perfect hit result
+                var statistics = beatmap.HitObjects
+                                        .Select(ho => ho.CreateJudgement().MaxResult)
+                                        .GroupBy(hr => hr, (hr, list) => (hitResult: hr, count: list.Count()))
+                                        .ToDictionary(pair => pair.hitResult, pair => pair.count);
+                perfectPlay.Statistics = statistics;
+
+                // calculate max combo
+                var difficulty = difficultyCache.GetDifficultyAsync(
+                    beatmap.BeatmapInfo,
+                    originalScore.Ruleset,
+                    originalScore.Mods,
+                    cancellationTokenSource.Token
+                ).GetResultSafely();
+                perfectPlay.MaxCombo = difficulty?.MaxCombo ?? originalScore.MaxCombo;
```

Comment:
falling back to the score's max combo looks questionable - there's no actual guarantee it is the actual max combo of the map, is there? if the map isn't locally available to compute the combo, then the tooltip shouldn't be displayed at all.